### PR TITLE
Adding ttl required setting to container config

### PR DESCRIPTION
**Pull request by zzmao**
Add `ttlRequired` to `Container`. 
Reference: #919

### CODE REVIEW

**Comment by lightningrob** [comment]
Where is the ttlRequired property used?  (What does it do?)

**Comment by zzmao** [comment]
It's a container level property, checking if a PUT should have ttl based on container's `ttlRequired` property. 

The checking part will be in next PR.

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java [hunk 1]
```diff
@@ -127,7 +139,7 @@ public void getBlobPropertiesVariedInputTest() throws Exception {
     verifyBlobPropertiesConstructionFailure(headers, RestServiceErrorCode.InternalServerError);
 
     // no failures.
-    // ttl missing. Should be infinite time by default.
+    // ttl missing. Should be infinite time in non ttlRequired container.
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: "should be" is misleading without the "by default" ("Should be infinite time by default in non ttlRequired container.")

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java [hunk 2]
```diff
@@ -139,6 +151,11 @@ public void getBlobPropertiesVariedInputTest() throws Exception {
     headers.put(RestUtils.Headers.TTL, JSONObject.NULL);
     verifyBlobPropertiesConstructionSuccess(headers);
 
+    // Post with valid ttl to ttlRequired container.
+    headers = new JSONObject();
+    setAmbryHeadersForPut(headers, "100", serviceId, Container.DEFAULT_PUBLIC_CONTAINER, contentType, ownerId);
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: the container is not the TTL required container here?

File: ambry-api/src/test/java/com.github.ambry/account/InMemAccountService.java [hunk 3]
```diff
@@ -176,11 +176,13 @@ private Container getRandomContainer(short accountId) {
     boolean refContainerPreviousEncryption = refContainerEncryption || TestUtils.RANDOM.nextBoolean();
     boolean refContainerCaching = TestUtils.RANDOM.nextBoolean();
     boolean refContainerMediaScanDisabled = TestUtils.RANDOM.nextBoolean();
+    boolean refContainerTtlRequired = TestUtils.RANDOM.nextBoolean();
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/account/InMemAccountService.java]: maybe this should not be random

File: ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java [hunk 4]
```diff
@@ -1144,8 +1145,8 @@ private void doRuntimeExceptionRouterTest(RestMethod restMethod) throws Exceptio
           break;
         case POST:
           JSONObject headers = new JSONObject();
-          setAmbryHeadersForPut(headers, Utils.Infinite_Time, !refContainer.isCacheable(), "test-serviceID",
-              "text/plain", "test-ownerId", refAccount.getName(), refContainer.getName());
+          setAmbryHeadersForPut(headers, referenceTtl, !refContainer.isCacheable(), "test-serviceID", "text/plain",
```
Comment (vgkholla) [inline ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java]: The `refContainer` is actually used in a lot of tests in this class and `InMemAccountService.createAndAddRandomAccount()` is also used in a lot of other classes. Left a comment in `InMemAccountService` but I don't think we should randomize that value but just set it to `false`. If tests want to change it, they will.

File: ambry-api/src/test/java/com.github.ambry/account/InMemAccountService.java [hunk 5]
```diff
@@ -176,13 +177,11 @@ private Container getRandomContainer(short accountId) {
     boolean refContainerPreviousEncryption = refContainerEncryption || TestUtils.RANDOM.nextBoolean();
     boolean refContainerCaching = TestUtils.RANDOM.nextBoolean();
     boolean refContainerMediaScanDisabled = TestUtils.RANDOM.nextBoolean();
-    boolean refContainerTtlRequired = TestUtils.RANDOM.nextBoolean();
     return new ContainerBuilder(refContainerId, refContainerName, refContainerStatus, refContainerDescription,
         accountId).setEncrypted(refContainerEncryption)
         .setPreviouslyEncrypted(refContainerPreviousEncryption)
         .setCacheable(refContainerCaching)
         .setMediaScanDisabled(refContainerMediaScanDisabled)
-        .setTtlRequired(refContainerTtlRequired)
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/account/InMemAccountService.java]: explicitly set this to `false` (if someone changes the default in `Container` in the future)

**Pull request merged**